aws-s3-bucket.yml

```diff
@@ -120,6 +119,12 @@ provision:
       details: Whether or not to store objects using a write-once-read-many (WORM) model using Amazon S3 Object Lock. (see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html).
       default: false
       prohibit_update: true
+    - field_name: ol_configuration_default_retention_enabled
+      type: boolean
+      details: |
+        Whether this bucket has an Object Lock `configuration` enabled
+        To enable Object Lock for a new bucket, see `ol_enabled`
+      default: null
```

Comment:
As discussed in a sync yesterday, but flagging here for visibility - I think ideally we should move towards default values being of the correct type.
i.e `bool` - `false`, rather than `null`, `string` - `""`. 

I know we had an issue with ENUMs where `""` would fail, but potentially we can add a default ENUM value of `DISABLE`, or some value to denote a setting where you can disable. 

Potentially with this particular example it's not an issue as once versioning is enabled you can no longer disable it, but generally we want to ensure that a user could revert the fields back to the default value, whereas using `null` stops this due to the CSB restriction.

Reply:
We remove the enum and treat the nullable value in a separate PR. Thank you very much, @jimbo459